packages/nx/src/project-graph/project-graph-builder.ts

```diff
@@ -145,6 +145,36 @@ export class ProjectGraphBuilder {
     }
   }
 
+  /**
+   * Add an explicit dependency from a file in source project to target project
+   */
+  addExternalNodeDependency(
```

Comment:
This should be the same function as `addDependency`.. we can modify that one to check if the projects are under `externalNodes`

Reply:
It's the plan of a bigger refactoring of that logic, but I don't want to include that here.

The current problem with `addDependency` is that it requires a file and adds the dependency to that file. We, later on, traverse all files to collect the project's dependencies. This causes some errors to be swallowed (e.g. importing project statically and lazily in the same file - we only keep dependency which we later map to `static`, instead of creating two dependencies - `static` and `dynamic`). There is an open issue for this assigned to me.